File: Python/ch3_4_氣泡排序實作.py
a = [3,2,4,5,1]
for i in range(1, 5):
  for j in range(5-i):
    if a[j] > a[j+1]:
      tmp = a[j]
      a[j] = a[j+1]
      a[j+1] = tmp
  print('第',i,'回合結果:', end='')
  for k in range(5):
    print(a[k], end='')
  print()

File: C/ch3-4-氣泡排序實作.c
#include <stdio.h>

int main(){
    int a[5] = {3,2,4,5,1};
    int i, j, tmp, k;

    for(i=1; i<5; i++){
        for(j=0; j<5-i; j++){
            if(a[j] > a[j+1]){
                tmp = a[j];
                a[j] = a[j+1];
                a[j+1] = tmp;
            }
        }
        printf("第%d回合結果:", i);
        for (k=0;k<5;k++)
            printf("%d ", a[k]);
        printf("\n");
    }

    return 0; 
}